src/Randomizer.CrossPlatform/Program.cs

```diff
@@ -0,0 +1,238 @@
+﻿using System.Diagnostics;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+using MSURandomizerLibrary.Models;
+using MSURandomizerLibrary.Services;
+using Randomizer.Data.Options;
+using Randomizer.Data.Services;
+using Randomizer.SMZ3.Generation;
+using Serilog;
+using Serilog.Events;
+
+namespace Randomizer.CrossPlatform;
+
+public static class Program
+{
+    public static void Main(string[] args)
+    {
+        var logLevel = LogEventLevel.Warning;
+        if (args.Any(x => x == "-v"))
+            logLevel = LogEventLevel.Information;
+
+        Log.Logger = new LoggerConfiguration()
+            .Enrich.FromLogContext()
+            .WriteTo.Console(logLevel)
+            .WriteTo.File(LogPath, rollingInterval: RollingInterval.Day, retainedFileCountLimit: 30)
+            .CreateLogger();
+
+        var services = new ServiceCollection()
+            .AddLogging(logging =>
+            {
+                logging.AddSerilog(dispose: true);
+            })
+            .ConfigureServices()
+            .BuildServiceProvider();
+
+        InitializeMsuRandomizer(services.GetRequiredService<IMsuRandomizerInitializationService>());
+
+        var optionsFile = new FileInfo("randomizer-options.yml");
+        var randomizerOptions = services.GetRequiredService<OptionsFactory>().LoadFromFile(optionsFile.FullName, optionsFile.FullName, true);
+        randomizerOptions.Save();
+
+        if (string.IsNullOrEmpty(randomizerOptions.GeneralOptions.Z3RomPath) || string.IsNullOrEmpty(randomizerOptions
+                                                                                 .GeneralOptions.SMRomPath)
```

Comment:
I'm not sure if this is GitHub's viewer fucking up or what, but this indentation looks... weird. Maybe move this to a separate method or something?

Reply:
Thanks for catching on that. I noticed that and meant to create a service for validating the source roms. Just implemented that for both apps. And yeah, usually Rider's automatic formatting isn't too bad, but it really went to town on this one. lol